libs/generic-view/store/src/lib/file-transfer/send-files.action.ts

```diff
@@ -83,114 +84,176 @@ export const sendFiles = createAsyncThunk<
-        const sendFileBaseDispatch = dispatch(
-          sendFileViaSerialPort({
-            destinationPath,
-            customDeviceId,
-            entitiesType,
-            file,
-          })
-        )
-
-        sendFileAbortController.abort = (
-          sendFileBaseDispatch as unknown as { abort: AbortController["abort"] }
-        ).abort
-
-        const { meta, payload } = await sendFileBaseDispatch
-
-        if (meta.requestStatus === "rejected" && meta.aborted) {
-          const error = new AppError(ApiFileTransferError.Aborted, "Aborted")
-          dispatch(
-            sendFilesError({
-              id: file.id,
-              error,
-            })
-          )
-          return rejectWithValue(error)
-        }
-
-        if (meta.requestStatus === "rejected") {
-          const error =
-            payload instanceof AppError
-              ? payload
-              : new AppError(ApiFileTransferError.Unknown)
+    const checkMtpAvailability = async () => {
+      const res = await dispatch(
+        getMtpSendFileMetadata({ destinationPath, customDeviceId })
+      )
+      const { meta, payload } = res
 
-          dispatch(
-            sendFilesError({
-              id: file.id,
-              error,
-            })
-          )
-        }
-        return
+      if (meta.requestStatus === "fulfilled" && payload !== undefined) {
+        mtpSendFileMetadata = payload as Omit<SendFileViaMTPPayload, "file">
+        dispatch(setFilesTransferMode(FilesTransferMode.Mtp))
+        return true
       }
+      return false
+    }
 
+    const mtpMonitor = setInterval(async () => {
+      const currentMode = selectFilesTransferMode(getState())
       if (
-        "path" in file &&
-        filesTransferMode === FilesTransferMode.Mtp &&
-        mtpSendFileMetadata !== undefined
+        currentMode === FilesTransferMode.SerialPort &&
+        switchToMtpCounter <= maxSwitchToMtpTries
       ) {
-        const sendFileBaseDispatch = dispatch(
-          sendFileViaMTP({
-            ...mtpSendFileMetadata,
-            file,
-          })
-        )
-
-        sendFileAbortController.abort = (
-          sendFileBaseDispatch as unknown as { abort: AbortController["abort"] }
-        ).abort
-
-        const { meta, payload } = await sendFileBaseDispatch
-
-        if (meta.requestStatus === "rejected" && meta.aborted) {
-          const error = new AppError(ApiFileTransferError.Aborted, "Aborted")
-          dispatch(
-            sendFilesError({
-              id: file.id,
-              error,
-            })
-          )
-          return rejectWithValue(error)
+        const isMtpAvailable = await checkMtpAvailability()
+        if (isMtpAvailable) {
+          console.log("MTP became available, switching mode.")
+          sendFileAbortController.abort()
+          switchToMtpCounter++
         }
+      }
+    }, 3_000)
+    let currentFileIndex = 0
+
+    const processFiles = async () => {
```

Comment:
I'd consider moving the `processFiles` to a separate action (async thunk). This would split the logic into smaller files and could help in handling _abort_ state using the native thunk's behavior instead of `wasAborted` variable.